cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolationDetector.java

```diff
@@ -246,10 +246,10 @@ protected void refreshBalancednessScore(Map<Boolean, List<String>> violatedGoals
   }
 
   protected Set<String> excludedTopics(ClusterModel clusterModel) {
-    return clusterModel.topics()
-        .stream()
-        .filter(topic -> _excludedTopics.matcher(topic).matches())
-        .collect(Collectors.toSet());
+    if (_excludedTopics.pattern().isEmpty()) {
+      return Collections.emptySet();
+    }
+    return Utils.getTopicNamesMatchedWithPattern(_excludedTopics, clusterModel.topics());
```

Comment:
`Utils.getTopicNamesMatchedWithPattern(...)` will also perform the empty pattern check and return an empty set. Can we eliminate repeated check for empty pattern?

Reply:
Same response as above

Reply:
Please see the response in the other comment.